Modules/_threadmodule.c

```diff
@@ -1551,6 +1551,40 @@ PyDoc_STRVAR(excepthook_doc,
 \n\
 Handle uncaught Thread.run() exception.");
 
+
+static PyObject*
+thread_set_name(PyObject *self, PyObject *args) {
```

Comment:
```
 % ./python.exe
Python 3.11.0a5+ (heads/bpo-412569/propagate-thread-name:9f9acf49be, Feb  5 2022, 17:15:00) [Clang 13.0.0 (clang-1300.0.29.3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import _thread
>>> _thread.set_name(123434, "x")
zsh: segmentation fault  ./python.exe
```

This function basically just writes to arbitrary memory, which is a security risk. I think the right fix would be to pass the thread state around as some sort of opaque object instead of a raw int.

Reply:
Good point. `set_name` is not meant to be called by anything else other than `Thread`, but it must be made safe nonetheless! Is there an established pattern for this sort of issue? I'm thinking of making an object, like `_PyThreadState_Reference` to pass from C to Python.